src/draw/drawings.js

```diff
@@ -42,35 +43,44 @@ function createPathSprites(path, cpt, hexColor, alpha) {
   });
 }
 
-/*
- * If tagpro.renderer.pathSprites has been initialized, iterate through each stored object and
- * remove it from the background. Used for removing the drawing of the previous path from the map.
- */
-function clearRects() {
-  if (tagpro.renderer.pathSprites) {
-    tagpro.renderer.pathSprites.forEach(sprite => {
-      tagpro.renderer.layers.background.removeChild(sprite);
-    });
+function createNonTraversableCellSprites(
+  traversableCells, cpt, notTraverableColor, alpha) {
+  tagpro.renderer.nonTraversableCellSprites = [];
+  const pixelsPerCell = PIXELS_PER_TILE / cpt;
+  for (let x = 0; x < traversableCells.length; x++) {
+    for (let y = 0; y < traversableCells[0].length; y++) {
+      if (!traversableCells[x][y]) { // if cell is non traversable
+        const rect = new PIXI.Graphics();
+        rect.beginFill(notTraverableColor).drawRect(
+          x * pixelsPerCell, // x coordinate
+          y * pixelsPerCell, // y coordinate
+          pixelsPerCell, // width
+          pixelsPerCell, // height
+        ).alpha = alpha;
+        tagpro.renderer.nonTraversableCellSprites.push(rect);
+      }
+    }
   }
 }
 
 /*
- * Iterates over each PIXI.Graphics object in tagpro.renderer.pathSprites and adds it to the
- * background
- */
-function drawRects() {
-  tagpro.renderer.pathSprites.forEach(sprite => {
-    tagpro.renderer.layers.background.addChild(sprite);
-  });
-}
-
-/*
  * Used to draw the bot's planned path. Takes in a reference to a list of cells returned by
  * getShortestPath() in helpers/path.js and the cpt used when calculating shortest path, and draws
  * the corresponding cells in green on the map.
  */
-export default function drawPlannedPath(path, cpt, hexColor = 0x00ff00, alpha = 0.25) {
-  clearRects(); // clear the previous path from the map
-  createPathSprites(path, cpt, hexColor, alpha); // create the PIXI.Graphics objecs we're drawing
-  drawRects(); // put the PIXI.Graphics objects on the map
+export function drawPlannedPath(path, cpt, visuals = true, hexColor = 0x00ff00, alpha = 0.25) {
+  clearSprites(tagpro.renderer.pathSprites); // clear the previous path from the map
+  if (visuals) {
+    createPathSprites(path, cpt, hexColor, alpha); // create the PIXI.Graphics objecs we're drawing
+  }
+  drawSprites(tagpro.renderer.pathSprites); // put the PIXI.Graphics objects on the map
+}
+
+export function drawNonTraversableCells(traversableCells, cpt, visuals = true,
+  notTraverableColor = 0xff8c00, alpha = 0.4) {
+  clearSprites(tagpro.renderer.nonTraversableCellSprites);
+  if (visuals) {
+    createNonTraversableCellSprites(traversableCells, cpt, notTraverableColor, alpha);
+  }
+  drawSprites(tagpro.renderer.nonTraversableCellSprites);
```

Comment:
and yeah, if visuals are off, it iterates over empty array. i wrote it this way since it's clean, and doesn't really hurt runtime. up for suggestions to improve readability though!